apps/core/tasks/auto_upgrade/runner.py

```
@@ -0,0 +1,290 @@
+            base_dir,
+            (
+                "Delegated upgrade launch failed "
+                f"(exit code {result.returncode}{details})"
+            ),
+        )
+        return None
+
+    return unit_name
+
+
+def _resolve_service_url(base_dir: Path) -> str:
```

Ruff (0.14.14)

[warning] Unused function argument: `base_dir`

(ARG001)